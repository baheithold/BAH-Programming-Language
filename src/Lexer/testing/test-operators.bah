# This file tests valid operators


# Binary operators
+
-
*
/
^
%
=
+=
-=
*=
/=
^=
%=
.

# Unary operators
++
--
!

# Comparators
==
!=
>
<
>=
<=

# Logical operators
and
or
xor
