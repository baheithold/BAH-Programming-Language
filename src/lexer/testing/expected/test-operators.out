PLUS_BINARY
MINUS_BINARY
TIMES_BINARY
DIVIDE_BINARY
POW_BINARY
MODULO_BINARY
ASSIGN_BINARY
PLUS_ASSIGN_BINARY
MINUS_ASSIGN_BINARY
TIMES_ASSIGN_BINARY
DIVIDE_ASSIGN_BINARY
POW_ASSIGN_BINARY
MODULO_ASSIGN_BINARY
DOT_BINARY
INCREMENT_UNARY
DECREMENT_UNARY
NEGATE_UNARY
EQUALS_COMPARATOR
NOT_EQUALS_COMPARATOR
GREATER_THAN_COMPARATOR
LESSER_THAN_COMPARATOR
GREATER_EQUALS_COMPARATOR
LESSER_EQUALS_COMPARATOR
AND
OR
XOR
END_OF_INPUT
